File: academic-info/src/main/kotlin/com/formula1/academicinfo/model/OptionalDiscipline.kt
package com.formula1.academicinfo.model
import javax.persistence.*

@Entity
@Table(name = "optional_discipline")
class OptionalDiscipline() {

    @Id
//    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    @Column(name = "optional_discipline_id", nullable = false)
    var oDisciplineId: Int = 0

    @Column(name = "max_attendants")
    var maxAttendants: Int = 0

    @Column(name = "current_attendants")
    var currentAttendants: Int = 0

    @Column(name = "approved")
    var approved: Boolean = false

    @OneToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinColumn(name = "optional_discipline_id")
    var optionalDiscipline: Discipline? = null

    @OneToMany(mappedBy = "optionalsSelectionDiscipline",
        fetch = FetchType.LAZY,
        cascade = [CascadeType.PERSIST])
    var optionalsSelections: MutableSet<OptionalsSelection> = mutableSetOf()

}

File: academic-info-golang/model/optional_discipline.go
package model

type OptionalDiscipline struct {
	OptionalDisciplineId int         `gorm:"primaryKey;column:optional_discipline_id;type:int"`
	MaxAttendants        int         `gorm:"column:max_attendants;type:int"`
	Approved             bool        `gorm:"column:approved;type:bool"`
	CurrentAttendants    int         `gorm:"column:current_attendants;type:int"`
	Discipline           *Discipline `gorm:"foreignKey:OptionalDisciplineId;associationForeignKey:DisciplineId"`
}

func (OptionalDiscipline) TableName() string {
	return "optional_discipline"
}
